### PR TITLE
tests: Add comprehensive unit tests for OrderService

### PR DESCRIPTION
**Pull request by JzomitiZumatti**
This commit adds unit tests to ensure the correct behavior of methods in the OrderService class. The tests cover various scenarios including creating an order, retrieving orders by ID, updating orders, adding products to orders, deleting products from orders, and deleting orders by ID. Each test validates the expected behavior of the corresponding method under different conditions, ensuring proper handling of edge cases and error scenarios. Additionally, Mockito is used to mock dependencies such as the OrderRepository, allowing isolated testing of the OrderService functionality. These tests provide comprehensive coverage to ensure the reliability and correctness of the OrderService implementation.

### CODE REVIEW

File: src/test/java/com/example/glovodb/service/OrderServiceTest.java [hunk 1]
```diff
@@ -1,5 +1,159 @@
...
+        Order initialOrder = Order.builder().id(TEST_ID).build();
+        when(orderRepository.findById(anyInt())).thenReturn(Optional.empty());
+        ResponseEntity<Order> response = orderService.updateOrder(TEST_ID, initialOrder);
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    }
+
+    @Test
+    public void addProductToOrderTest() {
+        Collection<Product> products = new ArrayList<>();
+        products.add(new Product());
```
Comment (IvashchenkoIvanIgorevich) [inline src/test/java/com/example/glovodb/service/OrderServiceTest.java]: You can simplify that by using `Collections.singletonList(new Product());` and past it to the products fields in builder

File: src/test/java/com/example/glovodb/service/OrderServiceTest.java
Comment (IvashchenkoIvanIgorevich) [inline src/test/java/com/example/glovodb/service/OrderServiceTest.java:1]: Please make refactoring by adding empty lines between Arrange, Act, and Assert.

**Review by IvashchenkoIvanIgorevich — APPROVED** [review]
Good work!

File: src/test/java/com/example/glovodb/service/OrderServiceTest.java (hunk 1 above)
Comment (JzomitiZumatti) [reply src/test/java/com/example/glovodb/service/OrderServiceTest.java]: I tried but got "java.lang.UnsupportedOperationException", because than I'm trying to add product to immutable collection.

**Review by ZarubeyAleksandr — APPROVED** [review]

File: src/test/java/com/example/glovodb/service/OrderServiceTest.java (hunk 1 above)
Comment (IvashchenkoIvanIgorevich) [reply src/test/java/com/example/glovodb/service/OrderServiceTest.java]: Yes, my bad, sorry.
